packages/desktop-client/src/hooks/usePreviewTransactions.ts

```diff
@@ -1,34 +1,66 @@
-import { useMemo } from 'react';
+import { useState } from 'react';
 
 import {
   type ScheduleStatuses,
   useCachedSchedules,
 } from 'loot-core/client/data-hooks/schedules';
+import { send } from 'loot-core/platform/client/fetch';
+import { ungroupTransactions } from 'loot-core/shared/transactions';
 import { type ScheduleEntity } from 'loot-core/types/models';
 
-export function usePreviewTransactions() {
+import { type TransactionEntity } from '../../../loot-core/src/types/models/transaction.d';
+
+export function usePreviewTransactions(
+  collapseTransactions: (ids: string[]) => void,
+) {
   const scheduleData = useCachedSchedules();
+  const [previousScheduleData, setPreviousScheduleData] =
```

Comment:
Can we use `usePrevious` here?

Reply:
I think `usePrevious` always returns the value on the previous render, whereas here we need the value from the last time `scheduleData` was changed. Maybe a name like `currentScheduleData` would be more accurate to the intent. What do you think of that name?